### ISSUE TITLE
[FEATURE] Requests metadata from the file name

### ISSUE THREAD

**Issue by hucurve** [stumpapp/stump#500]
I like the style of stump and would like to move my comic library here
An important problem, however, is that there is almost no corresponding file for the standard metadata in the ZIP file

I, however, followed the EHERENTAI naming convention to the letter
(years)[ Artist ] TITLE (Series)[ Language ] .

I think this can greatly facilitate the process of dealing with a large number of folders
Is also my hope that Stump can develop the function, I wish Stump better and better，thank U！

**Comment by aaronleopold** [comment]
Happy you like the style of Stump!

> An important problem, however, is that there is almost no corresponding file for the standard metadata in the ZIP file

I'm not sure what you mean, maybe I am just misunderstanding. Stump does collect metadata from a select set of files common in the community, see https://www.stumpapp.dev/guides/basics/books#metadata. Are you saying that your files, specifically, do not have metadata files inside of them?

> (years)[ Artist ] TITLE (Series)[ Language ]

I'm not necessarily opposed to supporting this, but it would be a medium-large effort to adjust the scanner to look for and match on it. I'll keep this in consideration, but can't make any timeline promises 🙂

**Comment by hucurve** [comment]
That's right, most of my files are comic strip packages, almost all of them are pictures, because no one will maintain and write metadata files. They look like these：
![image](https://github.com/user-attachments/assets/9b6d4972-213e-4c52-a7d3-ed77d80b2f72)
I've used pretty much all the self-hosting software available, and I think the most feature-rich is the desktop version of Calibre, but I gave up because it was too much work. I think some of the best features are (1) custom character scraping for author name, title, and so on, (2) author name mapping because authors can have multiple names, and (3) write metadata, custom edited metadata can be written to. Zip, . CBR compression package for easy archiving. I hope this helps developers think outside the box, but I wish Stump every success as a differentiated, self-hosted software, and I'd move quickly to add it if more function could be added. Best wishes.
